tests/test_detection.py

```diff
@@ -216,6 +216,30 @@ def checkExposure(original, doTempLocalBackground, doTempWideBackground):
         checkExposure(original, False, True)
         checkExposure(original, True, True)
 
+    def test_removeBadPixels(self):
+        """Test that if we set a NO_DATA region on top of a source,
+        One fewer sources is detected than when we don't do that."""
+        badMaskPlanes = ["NO_DATA",]
+        exposure_regular, schema, numX, numY, starSigma = self._create_exposure()
+        exposure_removed = exposure_regular.clone()
+
+        # Define a region on one test exposure we will set to NO_DATA, blocking out one source
+        region = lsst.geom.Box2I(exposure_removed.getXY0(), lsst.geom.Extent2I(15, 35))
```

Comment:
Huh, that is weird, because I swore it did cover a source when I was testing it locally! Thanks for checking. I needed to run Jenkins anyway still, as you wisely noted 😉